### ISSUE TITLE
Adam won't converge

### ISSUE THREAD

**Issue by devinbostIL** [OpenNMT/OpenNMT-py#343]
I think there might be a bug in the latest version of PyOpenNMT. 
I ran a model with the default settings for Adam, and my model didn't converge at all....

I ran it from:
`python train.py -data data/SSIS_Export_gen3_data_10-13-2017 -gpuid 0 -layers 4 -epochs 20 -optim adam -rnn_size 500 -save_model SSIS_Export_gen3_data_10-13-2017_adam_4layer_500neuron | tee data/SSIS_Export_gen3_data_10-13-2017_adam_4layer_500neuron_log_train.txt`

on my machine with Ubuntu 16.04 running my Titan Xp GPU. 
I noticed the same pattern where Adam wouldn't converge with a 2 layer model. 
However, with SGD, the model converged just fine.
Here are logs of the results of running the various commands. (I have clipped the output for all but one of the logs.)

I wondered if the learning rate for Adam is decaying way too quickly, but I didn't see any documentation on what the decay defaults are for Adam, and I would still expect to see the perplexity decrease for at least the very first iteration in even that case. 

[log_3_end_part_of_Adam_4_layer.txt](https://github.com/OpenNMT/OpenNMT-py/files/1388252/log_3_end_part_of_Adam_4_layer.txt)
[log_3_full_SSIS_Export_gen3_data_10-13-2017_adam_4layer_500neuron_40epochs_log_train.txt](https://github.com/OpenNMT/OpenNMT-py/files/1388253/log_3_full_SSIS_Export_gen3_data_10-13-2017_adam_4layer_500neuron_40epochs_log_train.txt)
[log_1_Adam_2layer.txt](https://github.com/OpenNMT/OpenNMT-py/files/1388254/log_1_Adam_2layer.txt)
[log_2_SGD_2layer.txt](https://github.com/OpenNMT/OpenNMT-py/files/1388255/log_2_SGD_2layer.txt)

**Comment by sebastianGehrmann** [comment]
Please use a small learning rate (~0.001) with adam. It should all work then.

**Comment by devinbostIL** [comment]
Isn't that what the default learning rate is supposed to be for Adam?

**Comment by sebastianGehrmann** [comment]
The default learning rate in OpenNMT is 1. You need to specify it via `-learning_rate`

**Comment by devinbostIL** [comment]
Maybe the documentation needs to be updated. 
"Starting learning rate. If adagrad/adadelta/adam is used, then this is the
global learning rate. Recommended settings: sgd = 1, adagrad = 0.1, adadelta =
1, adam = 0.001"

I read that like this:
"**Starting learning rate**. If adagrad/adadelta/adam is used, then this is the
global learning rate. Recommended settings: sgd = 1, adagrad = 0.1, adadelta =
1, **adam = 0.001**"

http://opennmt.net/OpenNMT-py/options/train/

**Comment by devinbostIL** [comment]
Setting the learning rate to 0.001 does seem to solve the problem. 
Should I expect any issues with the learning rate decay parameters without manually setting them also?

**Comment by sebastianGehrmann** [comment]
No, everything else should work as expected. (unless you don't want to have a decay)

**Comment by srush** [comment]
Thanks @devinbostIL, we will update the string.

**Comment by he1ght** [comment]
Oh, I got similar issue with this.
My result of adam's accuracy was too low and perplexity was too large.
Was this bug?  Is this fixed now?
